### PR TITLE
Bookstore: initialize project with components

### PR DESCRIPTION
**Pull request by Bornittah**
**Implementation**

- Initialise React app.
- Create a directory for your components.
- Add [React Router](https://reactrouter.com/web/guides/quick-start) and set two <Route>s and <Link>s for the app's navigation:
    - Books - the default view:
        - Should display the list of books (empty at this point but it should be ready for the data) with the Remove button (no funcionality yet).
            - Create a component called Book for displaying a single book (receive a title and an author as prop) and reuse it in a component that displays a list of books.
      - Should have a form for adding a book (no functionality yet).
          - Create form component with inputs for a title and an author 
    - Categories:
        - Should display a button "Check status" only.

### CODE REVIEW

File: src/components/Books.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  <div>
+    <h2>List of Books</h2>
+    <ul>
+      <li>
+        Fight Hunger by Nim
+        <Button value="Remove" />
+      </li>
+      <li>
+        Fight Hunger by Nim
+        <Button value="Remove" />
```
Comment (DammyShittu) [inline src/components/Books.jsx]: - [ ] This remove button should be in the Book component as that is where the remove event will be triggered and handled. Kindly move it to the component.

File: src/components/Books.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import Button from './shared/Button';
+
+const Books = () => (
+  <div>
+    <h2>List of Books</h2>
+    <ul>
+      <li>
+        Fight Hunger by Nim
+        <Button value="Remove" />
+      </li>
```
Comment (DammyShittu) [inline src/components/Books.jsx]: - [ ] Let's set up this component to render the Book component with the book data passed in as a prop. See example

  ```
  import Book from './Book';
  import Form from './Form';
  import Button from './shared/Button';
  
  const Books = () => (
    <div>
      <h2>List of Books</h2>
      <ul>
        <li>
          <Book title="first book" author="first author" />
        </li>
  .
  .
  .
  ```

File: src/components/Book.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const Book = () => (
+  <div>
+    <h2>Add abook</h2>
+    <ul>
+      <li>Music</li>
+    </ul>
+  </div>
+);
+
+export default Book;
```
Comment (DammyShittu) [inline src/components/Book.jsx]: - [ ] The Book component representing a single book should take in the book data as a prop and render it. Let's do that. Just as we have passed in the title and author as a prop earlier in the Books component, we can then accept the props here and render the title and author. In the end you should have something like this:

  ```
  import PropTypes from 'prop-types';
  import Button from './shared/Button';
  
  const Book = (props) => {
    const { title, author } = props;
    return (
      <div>
        <h2>{title}</h2>
        <ul>
          <li>{author}</li>
        </ul>
        <Button value="remove" />
      </div>
    );
  };
  
  Book.propTypes = {
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
  };

  export default Book;
  ```

  **NB:** You may need to install the `prop-types` package for prop validation and type checking

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Bornittah,

✔️ Use of Links for navbar

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/shared/NavBar.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const NavBar = () => (
+  <header>
+    <nav className="flex space-btn">
+      <ul className="navbar flex space-btn">
+        <li className="nav-link">
+          <Link to="/">BookStore CMS</Link>
+        </li>
+        <li className="nav-link">
+          <Link to="/">BOOKS</Link>
+        </li>
```
Comment (BenMukebo) [inline src/components/shared/NavBar.jsx:13]: - [ OPTIONAL] It is not necessary to have two links in the navbar which lead to the same page "Book page" 🤔  I think you only need to have Links in the navbar which lead to the booklist and categories locations 👍🏻

**Review by BenMukebo — APPROVED** [review]
### Hi @Bornittah, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**